### PR TITLE
Feat/convert ts d files

### PR DESCRIPTION
**Pull request by vdusart**
Moving to TS

## Related Issue
Conversion to TS according to #6392

### CODE REVIEW

File: overrides.d.ts [hunk 1]
```diff
@@ -2,3 +2,8 @@
 // errors.
 // TODO: create issue in Gatsby repo
 declare module "gatsby-plugin-image"
+
+declare module "*.yaml" {
+  const content: { [key: string]: any }
+  export default content
+}
```
Comment (pettinarip) [inline overrides.d.ts:9]: I think this is fine but I don't love it. I encourage you to define it a bit better (not using `any` and define it for each yaml file).

```ts
declare module "*developer-docs-links.yaml" {
  export interface DeveloperDocsLink {
    id: string
    to: string
    path: string
    description: string
    items: Array<DeveloperDocsLink>
  }
  const content: Array<DeveloperDocsLink>
  export default content
}

// then in DocsNav.tsx  you can use it like this
import docLinks, { DeveloperDocsLink } from "../data/developer-docs-links.yaml"
....
const getDocs = (links: Array<DeveloperDocsLink>): void => {
```

File: src/components/DocsNav.tsx [hunk 2]
```diff
@@ -78,9 +78,18 @@ const UppercaseSpan = styled.span`
...
+
+export interface IProps {
+  relativePath: string
+}
+
+const DocsNav: React.FC<IProps> = ({ relativePath }) => {
   // Construct array of all linkable documents in order recursively
-  const docsArray = []
+  const docsArray: DocsArrayProps[] = []
   const getDocs = (links) => {
```
[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/DocsNav.tsx:93]: (take a look at my other comment about defining the yaml file)

**Review by pettinarip — CHANGES_REQUESTED** [review]
Hey @vdusart Thanks for the work here. I left a couple of comments to improve the definitions and others to follow conventions. Gj 💪🏼

**Comment by vdusart** [comment]
Ok @pettinarip, I corrected all your comments, thanks for that. I will try to apply all of them for the next ones :wink:

**Comment by pettinarip** [comment]
@all-contributors please add @vdusart for code

**Pull request merged**